clients/fossid-webapp/src/main/kotlin/Extensions.kt

```diff
@@ -302,3 +309,54 @@ suspend fun FossIdRestService.createIgnoreRule(
             "apply_to" to scope.name.lowercase()
         )
     )
+
+suspend fun FossIdRestService.generateReport(
+    user: String,
+    apiKey: String,
+    scanCode: String,
+    reportType: ReportType,
+    selectionType: SelectionType
+): File {
+    val response = generateReport(
+        PostRequestBody(
+            "generate_report",
+            SCAN_GROUP,
+            user,
+            apiKey,
+            "scan_code" to scanCode,
+            "report_type" to reportType.name.lowercase(),
+            "selection_type" to selectionType.name.lowercase()
+        )
+    )
+
+    require(response.isSuccessful) {
+        """ ${response.code()} - ${response.message()}
+            ${response.errorBody()?.string()}
```

Comment:
Formatting looks a bit odd, why is there a single space after `"""`? Also, could you please add something about the context? Like "Report generation failed with ...".